{"name": "test_redirigir_facebook[chrome]", "status": "failed", "statusDetails": {"message": "Failed: El enlace a 'Bienestar Universitario UPT' no fue encontrado o no es clickable.", "trace": "driver = <selenium.webdriver.remote.webdriver.WebDriver (session=\"d185d84e6e207e4127a57773ebca15bb\")>\n\n    @allure.feature('Pruebas Home')\n    @allure.story('Verificación de enlaces en la página principal')\n    @pytest.mark.chrome\n    @pytest.mark.firefox\n    @pytest.mark.edge\n    def test_redirigir_facebook(driver):\n        with allure.step(\"Abrir la página principal\"):\n            driver.get(\"http://161.132.50.153/\")\n            sleep(3)\n    \n        with allure.step(\"Hacer clic en 'Bienestar Universitario UPT'\"):\n            try:\n>               link_facebook = WebDriverWait(driver, 20).until(\n                    EC.element_to_be_clickable((By.LINK_TEXT, \"Bienestar Universitario UPT\"))\n                )\n\nTeamWebQaUPT\\Features\\steps\\test_home_steps.py:44: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"d185d84e6e207e4127a57773ebca15bb\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x0000028F5F0FE5C0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55956ebb913a <unknown>\nE       #1 0x55956e89f5e0 <unknown>\nE       #2 0x55956e8eebe8 <unknown>\nE       #3 0x55956e8eee81 <unknown>\nE       #4 0x55956e9358c4 <unknown>\nE       #5 0x55956e913b4d <unknown>\nE       #6 0x55956e932d7d <unknown>\nE       #7 0x55956e9138c3 <unknown>\nE       #8 0x55956e8e16b3 <unknown>\nE       #9 0x55956e8e268e <unknown>\nE       #10 0x55956eb83b3b <unknown>\nE       #11 0x55956eb87ac1 <unknown>\nE       #12 0x55956eb70335 <unknown>\nE       #13 0x55956eb88642 <unknown>\nE       #14 0x55956eb5549f <unknown>\nE       #15 0x55956eba8038 <unknown>\nE       #16 0x55956eba8203 <unknown>\nE       #17 0x55956ebb7f8c <unknown>\nE       #18 0x7f95c14b0a94 <unknown>\nE       #19 0x7f95c153dc3c <unknown>\n\nC:\\Python312\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:105: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\ndriver = <selenium.webdriver.remote.webdriver.WebDriver (session=\"d185d84e6e207e4127a57773ebca15bb\")>\n\n    @allure.feature('Pruebas Home')\n    @allure.story('Verificación de enlaces en la página principal')\n    @pytest.mark.chrome\n    @pytest.mark.firefox\n    @pytest.mark.edge\n    def test_redirigir_facebook(driver):\n        with allure.step(\"Abrir la página principal\"):\n            driver.get(\"http://161.132.50.153/\")\n            sleep(3)\n    \n        with allure.step(\"Hacer clic en 'Bienestar Universitario UPT'\"):\n            try:\n                link_facebook = WebDriverWait(driver, 20).until(\n                    EC.element_to_be_clickable((By.LINK_TEXT, \"Bienestar Universitario UPT\"))\n                )\n                link_facebook.click()\n    \n                # Guardar el manejador de la pestaña original\n                original_window = driver.current_window_handle\n    \n                # Esperar que se abra una nueva pestaña\n                WebDriverWait(driver, 10).until(EC.number_of_windows_to_be(2))\n    \n                # Cambiar el foco a la nueva pestaña\n                for window_handle in driver.window_handles:\n                    if window_handle != original_window:\n                        driver.switch_to.window(window_handle)\n                        break\n    \n                with allure.step(\"Verificar la redirección a la página de Facebook\"):\n                    WebDriverWait(driver, 10).until(EC.url_contains(\"facebook.com\"))\n                    assert driver.current_url == \"https://www.facebook.com/ObunUPT/\", \"No se redirigió a la página de Facebook.\"\n    \n                # Cerrar la nueva pestaña\n                driver.close()\n    \n                # Regresar a la pestaña original\n                driver.switch_to.window(original_window)\n    \n            except TimeoutException:\n>               pytest.fail(\"El enlace a 'Bienestar Universitario UPT' no fue encontrado o no es clickable.\")\nE               Failed: El enlace a 'Bienestar Universitario UPT' no fue encontrado o no es clickable.\n\nTeamWebQaUPT\\Features\\steps\\test_home_steps.py:72: Failed"}, "steps": [{"name": "Abrir la página principal", "status": "passed", "start": 1732952342678, "stop": 1732952346625}, {"name": "Hacer clic en 'Bienestar Universitario UPT'", "status": "failed", "statusDetails": {"message": "Failed: El enlace a 'Bienestar Universitario UPT' no fue encontrado o no es clickable.\n", "trace": "  File \"C:\\Users\\Jean_\\proyecto-si8811a-2024-ii-u2-qa-pruebas-valverde-cano\\TeamWebQaUPT\\TeamWebQaUPT\\Features\\steps\\test_home_steps.py\", line 72, in test_redirigir_facebook\n    pytest.fail(\"El enlace a 'Bienestar Universitario UPT' no fue encontrado o no es clickable.\")\n  File \"C:\\Python312\\Lib\\site-packages\\_pytest\\outcomes.py\", line 178, in fail\n    raise Failed(msg=reason, pytrace=pytrace)\n"}, "start": 1732952346625, "stop": 1732952366882}], "parameters": [{"name": "driver", "value": "'chrome'"}], "start": 1732952342678, "stop": 1732952366895, "uuid": "0192c054-127a-41f2-95d7-920c40e2b4df", "historyId": "c0bb4c119ed4dab7821f38b09fc1410f", "testCaseId": "a7eb2beb4172c6f757bb1ead4dd174e2", "fullName": "TeamWebQaUPT.Features.steps.test_home_steps#test_redirigir_facebook", "labels": [{"name": "story", "value": "Verificación de enlaces en la página principal"}, {"name": "feature", "value": "Pruebas Home"}, {"name": "tag", "value": "edge"}, {"name": "tag", "value": "firefox"}, {"name": "tag", "value": "chrome"}, {"name": "parentSuite", "value": "TeamWebQaUPT.Features.steps"}, {"name": "suite", "value": "test_home_steps"}, {"name": "host", "value": "VALVERDE"}, {"name": "thread", "value": "31020-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "TeamWebQaUPT.Features.steps.test_home_steps"}]}